### PR TITLE
Add jsonpath as a dependency

### LINKED ISSUE #37: Port `jsonpath` fix from spring-cloud-stream-modules
As a developer, I'd like to add `jsonpath` as the dependency for all apps that supports `--expression` property. 

This [fix](https://github.com/spring-cloud/spring-cloud-stream-modules/pull/251) from `spring-cloud-stream-modules` repo needs ported over. 

Without the dependency, users are seeing the following error while using `jsonpath` in `--expression` property: 
```
2016-05-03T11:27:18.48-0700 [APP/0]      OUT 2016-05-03 18:27:18.486  WARN 20 --- [http.cart_add-1] ocessorConfiguration$HttpClientProcessor : Error in HTTP request
2016-05-03T11:27:18.48-0700 [APP/0]      OUT org.springframework.expression.spel.SpelEvaluationException: EL1022E:(pos 63): The function 'jsonPath' mapped to an object of type 'class org.springframework.expression.TypedValue' which cannot be invoked
```

### PR DESCRIPTION
**Pull request by sobychacko**
resolves #37

### CODE REVIEW

**Comment by sobychacko** [comment]
The apps that support --expression at the moment are processors and router. Introduced the dependencies at that level rather than doing it at the root pom.xml.

**Comment by melofred** [comment]
Guys, you're also missing the redis sink.. It supports SPeL for the attribute keyExpression . Might be missing others as well

**Comment by sobychacko** [comment]
Added the dependencies at the root level so that all apps can make use of those.

File: pom.xml [hunk 1]
```diff
@@ -19,6 +19,14 @@
 			<artifactId>spring-boot-configuration-processor</artifactId>
 			<optional>true</optional>
 		</dependency>
+      <dependency>
+        <groupId>com.jayway.jsonpath</groupId>
+        <artifactId>json-path</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>org.springframework.integration</groupId>
+        <artifactId>spring-integration-xml</artifactId>
```
Comment (ilayaperumalg) [inline pom.xml:28]: I am trying to understand why do we need `spring-integration-xml` here?

File: processor/spring-cloud-starter-stream-processor-transform/src/test/java/org/springframework/cloud/stream/app/transform/processor/TransformProcessorIntegrationTests.java [hunk 2]
```diff
@@ -69,10 +73,20 @@ public void test() {
 	@IntegrationTest("expression=payload.toUpperCase()")
 	public static class UsingExpressionIntegrationTests extends TransformProcessorIntegrationTests {
 
+		@Autowired
+		@Qualifier("&jsonPath")
+		private SpelFunctionFactoryBean jsonPathSpelFunctionFactoryBean;
+
+		@Autowired
+		@Qualifier("&xpath")
```
Comment (ilayaperumalg) [inline processor/spring-cloud-starter-stream-processor-transform/src/test/java/org/springframework/cloud/stream/app/transform/processor/TransformProcessorIntegrationTests.java:81]: ok, I see. This PR addresses `XPATH` as well.

File: pom.xml [hunk 3]
```diff
@@ -19,6 +19,14 @@
 			<artifactId>spring-boot-configuration-processor</artifactId>
 			<optional>true</optional>
 		</dependency>
+      <dependency>
```
Comment (ilayaperumalg) [inline pom.xml:22]: This needs to be formatted.

**Comment by ilayaperumalg** [comment]
LGTM

**Comment by ilayaperumalg** [comment]
Rebased and merged as 8395981eac5fbe3e6567c2cef159774e4e02b0c2 after fixing the formatting issue mentioned above.

**Pull request closed without merging**